Library/Homebrew/env_config.rb

```diff
@@ -42,6 +42,11 @@ module EnvConfig
                      "`brew cleanup` will start running `brew autoremove` periodically.",
         boolean:     true,
       },
+      HOMEBREW_AUTOREMOVE_INCLUDE_BUILD:         {
+        description: "If set, `brew autoremove` will parse build dependencies and not remove them. " \
+                     "If not set, `brew autoremove` only considers runtime dependencies.",
+        boolean:     true,
```

Comment:
Wait, shouldn't passing `--include-build` and setting `HOMEBREW_AUTOREMOVE_INCLUDE_BUILD` be equivalent? Because this docstring seems to suggest the opposite of the doc string for `--include-build`.

Though, I think we want a variable that does the opposite: if not set, `autoremove` retains its current behaviour. If set, `autoremove` omits build dependencies from removal.

Though, one way around this is to just explicitly `brew install` a build dependency (even if it is already installed) so that `brew` knows it has been installed on-request, preventing it from being `autoremove`d.